kiva/celiagg.py

```diff
@@ -89,8 +95,11 @@ def __init__(self, size, *args, **kwargs):
         self.pix_format = kwargs.get('pix_format', 'rgba32')
 
         shape = (self._height, self._width, 4)
+        buffer = np.zeros(shape, dtype=np.uint8)
         canvas_klass = pix_format_canvases[self.pix_format]
-        self.gc = canvas_klass(np.zeros(shape, dtype=np.uint8), bottom_up=True)
+        self.gc = canvas_klass(buffer, bottom_up=True)
+        marker_ren_klass = pix_format_marker_renderers[self.pix_format]
+        self.marker_gc = marker_ren_klass(buffer, bottom_up=True)
```

Comment:
Are there potential issues with keeping this synchronized with the main gc?  In particular, is there a risk of this holding an old buffer because something (eg. screen resize, code that is rummaging around in the internals of the GC when it shouldn't be, etc.) changed the original buffer.

I don't know about the performance implications, but if creating the marker gc is fast and lightweight enough it might make sense to create and dispose of as needed in `draw_marker_at_points`.

Or it may be that there isn't any way that the buffer can get changed, so this doesn't matter.

Reply:
The GC size never changes. If the parent widget is resized, a new GC is instantiated.